web/workflow.R

```diff
@@ -39,29 +39,36 @@ if (is.na(args[1])){
   settings <- read.settings(settings.file)
 }
 
+# Update/fix/check settings. Will only run the first time it's called, unless force=TRUE
+settings <- fix.deprecated.settings(settings, force=FALSE)
+settings <- addSecrets(settings, force=FALSE)
+settings <- update.settings(settings, force=FALSE)
+settings <- check.settings(settings, force=FALSE)
+
 # Check for additional modules that will require adding settings
 if("benchmark" %in% names(settings)){
   library(PEcAn.benchmark)
   settings <- papply(settings, read.settings.RR)
 }
 
 # Write pecan.CHECKED.xml
-settings <- write.settings(settings, outputfile = "pecan.CHECKED.xml")
+write.settings(settings, outputfile = "pecan.CHECKED.xml")
```

Comment:
> @bcow ... needed to put read.settings.RR in BEFORE settings were cleaned up so that she could leverage the checking/updating of settings

OK, will change order. 

> As per discussions on previous PR #1095,

As usual, apologies for not being up to speed. OK, so it sounds like you both agree with me that `write.settings` is a function we need and it should simply write settings to file. So I'm going to leave this function for use wherever we just want to save the settings. (It will be easy to extend later to other formats too.)

The other thing that happens is this update/fix/addsecrets/check thing. It sounds like the consensus is for all of these to be wrapped in a single function (`clean.settings`?), so that's what I'll do. 

But for future discussion, honestly that doesn't sound right to me. Obviously `fix.deprecated` and `addSecrets`, based on their names, should only need to be run once near the beginning of the workflow. Without looking now, I'm pretty sure that much of check.settings also just sets defaults and check things that can't reasonably be expected to change over the course of a workflow (like the DB)—again, actions that should only happen once, and it doesn't really matter in what order as long as it's early in the workflow. So it seems like for the most part the functionality of this `clean.settings` function can be thought of as a one-time sanitation operation for wayward settings.

If indeed there are also some parts of these functions that need to be used at specific times for specific purposes, I would suggest that they should be broken out into their own functions and used as needed. 

I'd also suggest that in the long run default-setting behavior be moved closer to the functions that rely on it, but that can be another discussion.